File: scripts/upload-documents.js
import fs from 'fs';
import FormData from 'form-data';
import fetch from 'node-fetch';

const API_URL = 'http://localhost:4010';
const EMAIL = 'office@local';
const PASSWORD = '1234';
const ZIP_PATH = process.argv[2] || '../../test-import.zip';

async function uploadDocuments() {
  try {
    // Login
    console.log('Logging in...');
    const loginRes = await fetch(`${API_URL}/auth/login`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ email: EMAIL, password: PASSWORD })
    });

    if (!loginRes.ok) {
      const error = await loginRes.text();
      throw new Error(`Login failed: ${error}`);
    }

    const { token } = await loginRes.json();
    console.log('Login successful!');

    // Upload file
    console.log(`Uploading ${ZIP_PATH}...`);
    const formData = new FormData();
    formData.append('file', fs.createReadStream(ZIP_PATH));

    const uploadRes = await fetch(`${API_URL}/imports/documents`, {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${token}`
      },
      body: formData
    });

    if (!uploadRes.ok) {
      const error = await uploadRes.text();
      throw new Error(`Upload failed: ${error}`);
    }

    const result = await uploadRes.json();
    console.log('\n=== Upload Results ===');
    console.log(`Processed: ${result.processed} documents`);
    if (result.errors && result.errors.length > 0) {
      console.log(`\nErrors (${result.errors.length}):`);
      result.errors.forEach((err, i) => {
        console.log(`  ${i + 1}. ${err}`);
      });
    } else {
      console.log('No errors!');
    }
  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  }
}

uploadDocuments();

File: scripts/upload-documents.py
#!/usr/bin/env python3
"""
Script להעלאת מסמכי DOCX דרך API
שימוש: python scripts/upload-documents.py <path-to-zip-file>
"""

import sys
import json
import requests
import os

API_URL = "http://localhost:4010"
EMAIL = "office@local"
PASSWORD = "1234"

def main():
    # Get zip file path
    zip_path = sys.argv[1] if len(sys.argv) > 1 else "../../test-import.zip"
    zip_path = os.path.abspath(zip_path)
    
    if not os.path.exists(zip_path):
        print(f"Error: File not found: {zip_path}")
        sys.exit(1)
    
    print("=== Document Upload Script ===\n")
    
    # Login
    print("Logging in...")
    login_data = {"email": EMAIL, "password": PASSWORD}
    try:
        login_response = requests.post(f"{API_URL}/auth/login", json=login_data)
        login_response.raise_for_status()
        token = login_response.json()["token"]
        print("Login successful!\n")
    except Exception as e:
        print(f"Login failed: {e}")
        sys.exit(1)
    
    # Upload file
    file_size = os.path.getsize(zip_path)
    print(f"Uploading: {zip_path}")
    print(f"File size: {file_size / 1024 / 1024:.2f} MB\n")
    
    try:
        with open(zip_path, 'rb') as f:
            files = {'file': (os.path.basename(zip_path), f, 'application/zip')}
            headers = {'Authorization': f'Bearer {token}'}
            response = requests.post(f"{API_URL}/imports/documents", files=files, headers=headers)
            response.raise_for_status()
            result = response.json()
        
        print("=== Upload Results ===")
        print(f"Processed: {result['processed']} documents")
        
        if result.get('errors') and len(result['errors']) > 0:
            print(f"\nErrors ({len(result['errors'])}):")
            for i, error in enumerate(result['errors'], 1):
                print(f"  {i}. {error}")
        else:
            print("No errors!")
        
        print("\nDone!")
        
    except Exception as e:
        print(f"\nUpload failed!")
        print(f"Error: {e}")
        if hasattr(e, 'response') and e.response is not None:
            print(f"Response: {e.response.text}")
        sys.exit(1)

if __name__ == "__main__":
    main()

